Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -1408,11 +1580,24 @@ private void HandleBrokerTime(object sender, IB.CurrentTimeUtcEventArgs e)
         /// 
         public IEnumerable<BaseData> GetNextTicks()
         {
+            Tick[] ticks;
+
             lock (_ticks)
             {
-                var copy = _ticks.ToArray();
+                ticks = _ticks.ToArray();
                 _ticks.Clear();
-                return copy;
+            }
+
+            foreach (var tick in ticks)
+            {
+                yield return tick;
+
+                if (_underlyings.ContainsKey(tick.Symbol))
+                {
+                    var underlyingTick = tick.Clone();
+                    underlyingTick.Symbol = _underlyings[tick.Symbol];
+                    yield return underlyingTick;
```

Comment:
That is correct. We send underlying data this way, assuming that enduser has no obligation to subscribe to underlying if he/she subscribed to its chains.